GameModel.md

```diff
@@ -0,0 +1,78 @@
+
+If the Worker sees that data are invalid and cannot be used for calculations, then there are various options to deal with the problem. One easy (and harsh for the Client) option is to allow the Worker to proceed as normal with the contract, and the Verifier and Arbiters should consider any results reported by the Worker as correct, and Client must give the full amount of Payment for this Work. Another option is to introduce a stage 1.5 which allows the Worker to back off from the contract after seing the data (if this is allowed without penalties, it makes possible a stalling attack by the Worker, but if we add a small penalty, it makes possible a flooding attack by the Client, so neither solution is perfect). Another option is to have Worker and Verifier both agree on poor quality of data, in which case the Client can receive refund of Payment minus a small penalty (while if Verifier does not agree with Worker on poor quality of data, then either Arbitration, or the Worker can back off from the contract with some penalty).
+
+
+Stage 3
+---
+_(not strictly in this order, can be inserted earlier)_
+
+Potential Verifiers make Initial Deposits (not necessa
+
+rily equal to v), declaring that they agree with the contract terms. The actual Verifier is chosen randomy among them and signs the contract for Work - this ends stage 3. Note thah the gathering and /or selection of Verifier could have been done at any time earlier; the timing is not critical, but it affects the length of time that the Verifier's stake is locked. There is a minor practical trade-off: choosing a Verifier late rather than early allows to minimize the time that the Verifier's stake is locked, however it creates more uncertainty whether the Verifier will be found in time.
+
+If the chosen Verifier fails to sign the contract by a certain time, then he is penalized with the size of his Initial Deposit. This penalty prevents the stalling attack. The money from penalty could be burnt or given to the Client and/or Worker as a compensation for wasting time. 
+
+
+Stage 4.
+---
+
+The Verifier receives the information how to calculate the Sample. The Verifier calculates the Sample and reports to the Client. Hence, the Client gets the information whether the Worker's results are considered correct by the Verifier - this ends stage 4. Note that it is not critical whether the Verifier received the Sample value calculated by the Worker. Both settings are possible: a) where the Verifier reports his own Sample result (without knowing the Worker's Sample result) and b) where the Verifier reports only a binary agreement or disagreement with the Worker's Sample result. However, the setting (a) is better, as it makes collusion more costly.
```

Comment:
Well, in this way we can say that there is some Worker that is ordered to re-do 10% of the work of some other Worker and there is some agent (Verifier or Client) that needs to produce a binary check are these results equal. The question is that: can we put this obligation on the Client, not Verifier? Would it distort its incentives?

Reply:
In my opinion, the one who produces the Sample calculations check should also be the one who reports whether it agrees with the initial Worker's result. Or (even better) this can be done automatically by comparing the two values. The Client, of course, can easily understand whether two values, reported to him, agree or not - however he should not be the one who reports this to the network. Adding more agents (including the Client) to the mix would complicate incentives and create "broken phone" situations, where information changes as it passes through people and it is hard to find who has modified it.